src/ddspdrum/module.py

```diff
@@ -235,3 +236,166 @@ def play(self):
         amp = self.control_to_sample_rate(self.get_envelope())
         signal = self.fix_length(self.get_audio(), len(amp))
         return amp * signal
+
+
+class SVF(SynthModule):
+    """
+    A State Variable Filter that can do lowpass, highpass, bandpass, and bandreject
+    filtering. Allows modulation of the cutoff frequency and an adjustable resonance
+    parameter. Can self-oscillate to make a sine / cosine wave.
+    """
+
+    def __init__(
+        self,
+        mode: str = "LPF",
+        cutoff: float = 1000,
+        resonance: float = 0.707,
+        self_oscillate: bool = False,
+        sample_rate: int = SAMPLE_RATE,
+    ):
+        super().__init__()
+        self.sample_rate = sample_rate
+        self.mode = mode
+        self.cutoff = cutoff
+        self.resonance = resonance
+        self.self_oscillate = self_oscillate
+
+    def __call__(
+        self,
+        audio: np.ndarray,
```

Comment:
Do we want to just call this `signal` everywhere? Or does it make sense to distinguish between `signal` and `audio`?